src/modules/simple_modules.py

```diff
@@ -103,20 +103,49 @@ def __init__(self, pooler, classifier):
         self.classifier = classifier
 
     def forward(self, sent, mask, idxs=[]):
-        """ Assumes batch_size x seq_len x d_emb """
+        """
+        This class applies some type of pooling to get a fixed-size vector,
+            possibly extracts some specific representations from the input sequence
+            and concatenates those reps to the overall representations,
+            then passes the whole thing through a classifier.
+
+        args:
+            - sent (FloatTensor): sequence of hidden states representing a sentence
+            Assumes batch_size x seq_len x d_emb.
+            - mask (FloatTensor): binary masking denoting which elements of sent are not padding
+            - idxs (List[LongTensor]): list of indices of to extract from sent and
+                concatenate to the post-pooling representation.
+                For each element in idxs, we extract all the non-pad (0) representations, pool,
+                and concatenate the resulting fixed size vector to the overall representation.
+
+        returns:
+            - logits (FloatTensor): logits for classes
+        """
+
         emb = self.pooler(sent, mask)
 
         # append any specific token representations, e.g. for WiC task
         ctx_embs = []
-        for idx in [i.long() for i in idxs]:
+        for idx in idxs:
             if len(idx.shape) == 1:
                 idx = idx.unsqueeze(-1)
             if len(idx.shape) == 2:
```

Comment:
Verify that this case still works, with some demo experiments and some debug printing.

Reply:
(Not all of these changed cases will be touched by demo.conf, IIRC.)

Reply:
Hmm I guess I was unclear / lazily programming. These are not cases so much as checks that the `idx` tensor is the shape we need it to be (3-dimensional, matching `sent`).